tests/soak/soakgoclient/soakclient_transaction_test.go

```diff
@@ -0,0 +1,267 @@
+                                        // But in this example we choose to
+                                        // terminate the application if all
+                                        // brokers are down.
+                                        fmt.Fprintf(os.Stderr,
+                                                    "kafka.Error: %v: %v\n",
+                                                    e.Code(),
+                                                    e)
+                                        if e.Code() == kafka.ErrAllBrokersDown {
+                                                run = false
+                                        }
+
+                                default:
+                                        t.Logf("Ignored %s\n", e)
+                        }
+                }
+        }
+        PrintConsumerStatus()
+        c.Close()
+}
+
+
+func produceMessage (t *testing.T,
+                    e *kafka.Message,
+                    p kafka.Producer) {
+        encodingTime, err := time.Now().GobEncode()
+        if err != nil {
+                t.Logf("Failed to get MarshalText %s\n", err)
+        }
+
+        err = p.Produce(&kafka.Message{
+                TopicPartition: kafka.TopicPartition{
+                        Topic: &OUT_TOPIC,
+                        Partition: kafka.PartitionAny},
+                Value: []byte(string(e.Value)),
+                Headers: []kafka.Header{{Key: "time",
+                                         Value: encodingTime}},
+              }, nil)
+        // Wait for delivery report
+        ev := <-p.Events()
+        switch e := ev.(type) {
+                case *kafka.Message:
+                        ProducerDeliveryCheck(e, OUT_TOPIC)
+
+                case *kafka.Stats:
+                        HandleStatsEvent(t, e, OUT_TOPIC)
+
+                default:
+                        t.Logf("Ignored event: %s\n", ev)
+        }
+}
+
+
+func commitTransaction (t *testing.T,
+                       partitionPosition []kafka.TopicPartition,
+                       p kafka.Producer,
+                       c kafka.Consumer,
+                       transactionCommitNum int) int {
+        fmt.Printf("=== Committing transaction===\n")
+
+        cgmd, err := c.GetConsumerGroupMetadata()
+        err = p.SendOffsetsToTransaction(nil, partitionPosition, cgmd)
+        if err != nil {
+                t.Fatalf("Failed to SendOffsetsToTransaction %s\n", err)
+        }
+        if (transactionCommitNum % 100 != 0) {
+                transactionCommitNum += 1
+                err = p.CommitTransaction(nil)
+                if err != nil {
+                        DatadogIncrement(PRODUCER_TRANSACTION_COMMIT_FAILED, 1)
+                        t.Fatalf("CommitTransaction() failed: %s\n", err)
+                } else {
+                        DatadogIncrement(PRODUCER_TRANSACTION_COMMIT_SUCCEED, 1)
+                }
+       } else {
+               err = p.AbortTransaction(nil)
```

Comment:
The consumer needs to rewind its position to the pre-transaction offset, i.e., the committed offset.
I think you can do this by simply doing something like:
```go
   assignment := c.Assignment()
   for _, tp = range assignment {
       tp.Offset = kafka.OffsetStored  // the committed offset
       err := c.Seek(tp)
   }
```